### ISSUE TITLE
Lint/UnneededDisable false positive after auto-correct of other cops

### ISSUE THREAD

**Issue by rrosenblum** [bbatsov/rubocop#2003]
I noticed that `Lint/UnneededDisable` will produce a false positive after another cop has auto-corrected the source and added or removed lines of code. It looks like this is being caused by `Lint/UnneededDisable` being told to check for the offense on a line that does not correspond to the disable comment anymore. 

Example:
```ruby
# test.rb
a = 1 # rubocop:disable Lint/UselessAssignment
```
run `rubocop -D -a test.rb`
```
Offenses:

test.rb:1:1: C: [Corrected] Style/Encoding: Missing utf-8 encoding comment.
a = 1 # rubocop:disable Lint/UselessAssignment
^
test.rb:1:7: W: Lint/UnneededDisable: Unnecessary disabling of Lint/UselessAssignment.
a = 1 # rubocop:disable Lint/UselessAssignment
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 1 offense corrected
```
run `rubocop -D -a test.rb` again
```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

I am not sure of how to fix this issue at the moment.

**Comment by alexdowad** [comment]
Can't reproduce this. Do you have a custom `.rubocop.yml`?

**Comment by rrosenblum** [comment]
This happens to me both using a custom `.rubocop.yml` and when using the default settings. I reproduced it against master.

**Comment by alexdowad** [comment]
Is the test file exactly as shown above?

**Comment by rrosenblum** [comment]
Yes, the test file is exactly as shown above. I have been able to reproduce the bug in every version of RuboCop since `Lint/UnneededDisable` was introduced on multiple machines running multiple versions of Ruby. 

The test file above is relying on `Style/Encoding` being enabled (off by default, but enabled in the RuboCop project).

It seems as though `Lint/UnneededDisable` gets its list of disabled lines before auto-correct is run. During auto-correct `a = 1 # rubocop:disable Lint/UselessAssignment` is moved down 1 line. `Lint/UnneededDisable` on the line that originally contained the `UselessAssignment`, but now no longer does.

**Comment by jonas054** [comment]
I see what the problem is. Solution coming up.

**Comment by alexdowad** [comment]
> I see what the problem is.

Nice! Looking forward to seeing it.
